# HG changeset patch
# User Kyle Guinn <elyk03@gmail.com>
# Date 1389629400 28800
#      Mon Jan 13 08:10:00 2014 -0800
# Branch stable
# Node ID 975480e1be4bbfa3eaf736aed702291e93c80ec6
# Parent  91805d8ab62faf28f62bfa0e4c961301bb88d1c6
configure.ac: Add AMD_CPPFLAGS to UMFPACK tests (bug #41209).

* configure.ac: Add AMD_CPPFLAGS to UMFPACK tests (bug #41209).

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1531,17 +1531,19 @@ if test -z "$UMFPACK_LIBS"; then
   fi
   LIBS="$save_LIBS"
 fi
 
 ## Test features of the installed UMFPACK library
 
 if test -n "$UMFPACK_LIBS"; then
   ## SuiteSparse >= 4.0 needs additional link library for SuiteSparse_time()
+  save_CPPFLAGS="$CPPFLAGS"
   save_LIBS="$LIBS";
+  CPPFLAGS="$UMFPACK_CPPFLAGS $AMD_CPPFLAGS $CPPFLAGS"
   LIBS="$UMFPACK_LIBS $AMD_LDFLAGS $AMD_LIBS $BLAS_LIBS $FLIBS $LIBS"
   xtra_libs=
   OCTAVE_UMFPACK_NEED_SUITESPARSE_TIME
   if test $octave_cv_umfpack_need_suitesparse_time = yes; then
     AC_CHECK_LIB([rt], [clock_gettime], [xtra_libs="-lrt"], [xtra_libs=])
     ## FIXME: This library list is only accurate for Linux, Mac OS X.
     ##        Possibly need other library names for MinGW, Cygwin.
     AC_SEARCH_LIBS([SuiteSparse_time],
@@ -1554,23 +1556,27 @@ if test -n "$UMFPACK_LIBS"; then
       no)
         UMFPACK_LIBS=
         AC_MSG_WARN([UMFPACK library found but is missing SuiteSparse_time functionality.])
         AC_MSG_WARN([UMFPACK library will be disabled.])
       ;;
     esac
   fi
   LIBS="$save_LIBS"
+  CPPFLAGS="$save_CPPFLAGS"
 
   ## Check for UMFPACK separately split complex matrix and RHS.
   if test -n "$UMFPACK_LIBS"; then
+    save_CPPFLAGS="$CPPFLAGS"
     save_LIBS="$LIBS";
+    CPPFLAGS="$UMFPACK_CPPFLAGS $AMD_CPPFLAGS $CPPFLAGS"
     LIBS="$UMFPACK_LIBS $CHOLMOD_LDFLAGS $CHOLMOD_LIBS $AMD_LDFLAGS $AMD_LIBS $COLAMD_LDFLAGS $COLAMD_LIBS $LAPACK_LIBS $BLAS_LIBS $FLIBS $LIBS $xtra_libs"
     OCTAVE_UMFPACK_SEPARATE_SPLIT
     LIBS="$save_LIBS"
+    CPPFLAGS="$save_CPPFLAGS"
   fi
 fi
 
 ### Check for ARPACK library.
 
 save_LIBS="$LIBS"
 LIBS="$LAPACK_LIBS $BLAS_LIBS $FLIBS $LIBS"
 OCTAVE_CHECK_LIB([arpack], ARPACK,
